packages/backend/src/open_payments/payment/outgoing/model.ts

```diff
@@ -63,6 +66,30 @@ export class OutgoingPayment
     return this.quote.receiveAmount
   }
 
+  private grantSpentDebitAmountValue?: bigint
+  public get grantSpentDebitAmount(): Amount {
+    return {
+      value: this.grantSpentDebitAmountValue || BigInt(0),
+      assetCode: this.asset.code,
+      assetScale: this.asset.scale
+    }
+  }
+  public set grantSpentDebitAmount(amount: Amount) {
+    this.grantSpentDebitAmountValue = amount.value
+  }
+
+  private grantSpentReceiveAmountValue?: bigint
+  public get grantSpentReceiveAmount(): Amount {
+    return {
+      value: this.grantSpentReceiveAmountValue || BigInt(0),
+      assetCode: this.asset.code,
+      assetScale: this.asset.scale
```

Comment:
same comment as in the service file. The asset code may be different than the payment itself.

Reply:
I guess this means we dont have any sort of default for the `grantSpentReceiveAmount` since this depends on being created with a grant limit. So we simply cant return `grantSpentReceiveAmount` for outgoing payments created in this case.

I guess that's fine... it just seems incongruent with the `grantSpentDebitAmount`, which we can return a default value for (0 with outgoing payment's `assetCode` and `assetScale`). I guess maybe that's OK, but it seems non-obvious from the caller's perspective. 

I guess we could:
- just leave as-is
- not return any defualt/null version  of `grantSpentDebitAmount` either
- return something like `{value: 0, assetCode: "", assetScale: -1 }` for both `grantSpentDebitAmount` and `grantSpentReceiveAmount` in this case

Reply:
or maybe we can get it from somewhere other than `grant.limits`...

Reply:
What if instead of storing `grantSpentReceiveAmountValue: bigint` and `grantSpentDebitAmountValue: bigint`, we just allow setting the whole `Amount`? Otherwise if not set, it just gets returned as undefined?

Reply:
I've changed it set the whole amount locally already so in full agreement there.

After sitting on this for a bit, I think `undefined` and then not returning them in the `/POST /outgoing-payment` seems fine. I guess my default expectation is to not include/exclude properties when they're in a sort of null state but in this case I think it may be the most straightforward.

Reply:
updated to above ^ [710897d](https://github.com/interledger/rafiki/pull/2753/commits/710897d194c4021316c4d13ae52e6415afe3222e)